internal/providers/pluginfw/resources/clusters/resource_library.go

```diff
@@ -0,0 +1,237 @@
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var libraryTfSDK LibraryExtended
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &libraryTfSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var libGoSDK compute.Library
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, libraryTfSDK, &libGoSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	installLib := compute.InstallLibraries{
+		Libraries: []compute.Library{libGoSDK},
+	}
+	req.Plan.GetAttribute(ctx, path.Root("cluster_id"), &installLib.ClusterId)
+	err := w.Libraries.Install(ctx, installLib)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to install library", err.Error())
+		return
+	}
+	waitParams := compute.Wait{
+		ClusterID: installLib.ClusterId,
+		IsRunning: true,
+	}
+	libraryRep := libGoSDK.String()
+	installedLib := LibraryExtended{}
+
+	resp.Diagnostics.Append(readLibrary(ctx, w, waitParams, libraryRep, &installedLib)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, installedLib)...)
+}
+
+func (r *LibraryResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var libraryTfSDK LibraryExtended
+	resp.Diagnostics.Append(req.State.Get(ctx, &libraryTfSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var libGoSDK compute.Library
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, libraryTfSDK, &libGoSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	clusterId := libraryTfSDK.ClusterId.ValueString()
+	libraryRep := libGoSDK.String()
+	installedLib := LibraryExtended{}
+	waitParams := compute.Wait{
+		ClusterID: clusterId,
+		IsRefresh: true,
+	}
+
+	resp.Diagnostics.Append(readLibrary(ctx, w, waitParams, libraryRep, &installedLib)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, installedLib)...)
+}
+
+func (r *LibraryResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+}
+
+func (r *LibraryResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var libraryTfSDK LibraryExtended
+	resp.Diagnostics.Append(req.State.Get(ctx, &libraryTfSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	clusterID := libraryTfSDK.ClusterId.ValueString()
+	var libGoSDK compute.Library
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, libraryTfSDK, &libGoSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	libraryRep := libGoSDK.String()
+	_, err := clusters.StartClusterAndGetInfo(ctx, w, clusterID)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to start and get cluster", err.Error())
+		return
+	}
```

Comment:
Wild that you need to start the cluster to list its libraries... again, seems like this is part of the original implementation.